src/fastpm.c

```diff
@@ -999,7 +999,17 @@ usmesh_ready_handler(FastPMUSMesh * mesh, FastPMLCEvent * lcevent, struct usmesh
     /* subsample */
     FastPMParticleMaskType * mask = fastpm_memory_alloc(lcevent->p->mem,
         "SubsampleMask", lcevent->p->np * sizeof(mask[0]), FASTPM_MEMORY_FLOATING);
-    fastpm_store_fill_subsample_mask(lcevent->p, CONF(prr->lua, particle_fraction), mask);
+    double particle_fraction = CONF(prr->lua, particle_fraction);
+    double a_mean = 0.5 * (lcevent->af + lcevent->ai);
```

Comment:
probably a bug

Reply:
Done. Using a_mean was problematic and we switched to use a per particle fraction.